synapse/handlers/room.py

```diff
@@ -464,15 +464,20 @@ async def clone_existing_room(
         # the room has been created
         # Calculate the minimum power level needed to clone the room
         event_power_levels = power_levels.get("events", {})
+        assert isinstance(event_power_levels, dict)
```

Comment:
I think we do (now) validate power levels events from the C-S API, but you could always get a weird event over federation. So I think it needs to be validated here.

I don't know if an assertion is the right thing to do though since this could stop you from upgrading a room with a funky event, which might not make sense.